zebra-state/src/service/finalized_state/disk_format/transparent.rs

```diff
@@ -660,14 +688,16 @@ impl IntoDisk for AddressBalanceLocation {
 
 impl FromDisk for AddressBalanceLocation {
     fn from_bytes(disk_bytes: impl AsRef<[u8]>) -> Self {
-        let (balance_bytes, address_location_bytes) =
-            disk_bytes.as_ref().split_at(BALANCE_DISK_BYTES);
+        let (balance_bytes, rest) = disk_bytes.as_ref().split_at(BALANCE_DISK_BYTES);
+        let (received_bytes, address_location_bytes) = rest.split_at(BALANCE_DISK_BYTES);
 
         let balance = Amount::from_bytes(balance_bytes.try_into().unwrap()).unwrap();
         let address_location = AddressLocation::from_bytes(address_location_bytes);
+        let received = Amount::from_bytes(received_bytes.try_into().unwrap_or_default()).unwrap();
 
         let mut address_balance_location = AddressBalanceLocation::new(address_location);
         *address_balance_location.balance_mut() = balance;
+        *address_balance_location.received_mut() = received;
 
         address_balance_location
     }
```

Comment:
I haven't fully reviewed this yet, but one thing I'm not sure about: don't we need to jump through [hoops](https://zebra.zfnd.org/dev/state-db-upgrades.html#adding-a-column-family) to make sure everything works during a database upgrade? Won't this code break trying to read values in the old format, or am I missing something?

Reply:
> Won't this code break trying to read values in the old format, or am I missing something?

The `FromDisk` impl for `AddressBalanceLocation` uses `received_bytes.try_into().unwrap_or_default()` to read the received bytes or default to 0 if it's in the old format.